File: LEETCODE/LEETCODE-CPP/217.cpp
//217. Contains Duplicate
//https://leetcode.com/problems/contains-duplicate/

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        unordered_map<int,int> m;
        int n=nums.size();
        
        for(int i=0;i<n;i++)
            m[nums[i]]++;
        
        for(auto i:m)
            if(i.second>1)
                return true;
        
        return false;
        
    }
};

File: LEETCODE/LEETCODE-PYTHON/217.py
# 217. Contains Duplicate
# https://leetcode.com/problems/contains-duplicate/description/

# Method-1 using set
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        unique_set=set()

        for i in nums:
            if i in unique_set:
                return True    
            unique_set.add(i)
                
        return False
        